oidc-service-impl/src/main/java/io/mosip/esignet/services/OAuthServiceImpl.java

```diff
@@ -202,23 +205,24 @@ private void validatePKCE(ProofKeyCodeExchange proofKeyCodeExchange, String code
             throw new EsignetException(ErrorConstants.PKCE_FAILED);
     }
 
-    private void authenticateClient(TokenRequest tokenRequest, ClientDetail clientDetail) throws EsignetException {
+    private void authenticateClient(TokenRequest tokenRequest, ClientDetail clientDetail,boolean isV2) throws EsignetException {
         switch (tokenRequest.getClient_assertion_type()) {
             case JWT_BEARER_TYPE:
-                validateJwtClientAssertion(clientDetail.getId(), clientDetail.getPublicKey(), tokenRequest.getClient_assertion());
+                validateJwtClientAssertion(clientDetail.getId(), clientDetail.getPublicKey(), tokenRequest.getClient_assertion(),
+                        isV2? (String) oauthServerDiscoveryMap.get("token_endpoint") :discoveryIssuerId+"/v1/esignet/oauth/token");
```

Comment:
@pvsaidurga 
mosip.esignet.discovery.issuer-id=${mosip.esignet.domain.url}${server.servlet.path}

so, discoveryIssuerId already has the servlet path, so here it should be only "/oauth/token"